akka-stream-tests/src/test/scala/akka/stream/scaladsl/QueueSinkSpec.scala

```diff
@@ -112,5 +114,20 @@ class QueueSinkSpec extends AkkaSpec {
       queue.pull().onFailure { case e ⇒ e.isInstanceOf[IllegalStateException] should ===(true) }
     }
 
+    "keep on sending even after the buffer has been full" in assertAllStagesStopped {
+      val (probe, queue) = Source(1 to 20)
+        .alsoToMat(Flow[Int].take(15).watchTermination()(Keep.right).to(Sink.ignore))(Keep.right)
+        .toMat(Sink.queue())(Keep.both)
+        .run()
+      probe.futureValue should ===(akka.Done)
```

Comment:
I misunderstood the way probe was used in #19654, and, based on @rkuhn's suggestion from conversation with @johanandren, I believe this is the correct way to test buffer filling up.

Buffer can have at most 15 upstream elements (not stream close/failure indicators) in it, based on the condition in `onPull` method.